clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/github/GitHubArtifactCredentials.java

```diff
@@ -93,18 +92,40 @@ private String credentialsFromFile(String filename) {
 
   public InputStream download(Artifact artifact) throws IOException {
     HttpUrl.Builder metadataUrlBuilder = HttpUrl.parse(artifact.getReference()).newBuilder();
-    metadataUrlBuilder.addQueryParameter("ref", artifact.getVersion());
+    String version = artifact.getVersion();
+    if (StringUtils.isEmpty(version)) {
+      log.info("No version specified for artifact {}, using 'master'.", version);
+      version = "master";
+    }
+
+    metadataUrlBuilder.addQueryParameter("ref", version);
     Request metadataRequest = requestBuilder
       .url(metadataUrlBuilder.build().toString())
       .build();
-    Response metadataResponse = okHttpClient.newCall(metadataRequest).execute();
+
+    Response metadataResponse;
+    try {
+      metadataResponse = okHttpClient.newCall(metadataRequest).execute();
+    } catch (IOException e) {
+      throw new FailedDownloadException("Unable to determine the download URL of artifact " + artifact + ": " + e.getMessage(), e);
```

Comment:
What's the `Artifact.toString()` look like? Will it be a nice human readable representation?